oak_client/src/attestation.rs

```diff
@@ -109,50 +134,109 @@ impl rustls::ServerCertVerifier for RemoteAttestationVerifier {
-        // Only a certificate created by the Attestation Proxy is required.
-        if certs.len() != 1 {
-            return Err(rustls::TLSError::General(format!(
-                "Only a single certificate must be provided, provided {}",
-                certs.len()
-            )));
-        }
-        let certificate = &certs[0].0;
-
-        // Parse X.509 certificate encoded using DER format.
-        // https://tools.ietf.org/html/rfc7468
-        let (_, parsed_certificate) = parse_x509_certificate(&certificate).map_err(|error| {
-            rustls::TLSError::General(format!("Couldn't parse certificate: {}", error))
-        })?;
-
-        // Extract X.509 TEE extension.
-        let extensions = parsed_certificate.extensions();
-        let tee_extension = extensions.get(&TEE_EXTENSION_OID).ok_or_else(|| {
-            rustls::TLSError::General("Couldn't find X.509 TEE extension".to_string())
-        })?;
-        let tee_extension_value = parse_der(tee_extension.value)
-            .map_err(|error| {
-                rustls::TLSError::General(format!(
-                    "Couldn't parse X.509 TEE extension: {:?}",
+        for certificate in certs.iter() {
+            // Parse X.509 certificate encoded using DER format.
+            // https://tools.ietf.org/html/rfc7468
+            let (_, parsed_certificate) =
+                parse_x509_certificate(&certificate.0).map_err(|error| {
+                    let error =
+                        rustls::TLSError::General(format!("Couldn't parse certificate: {}", error));
+                    warn!("{:?}", error);
                     error
-                ))
-            })?
-            .1
-            .as_str()
-            .map_err(|error| {
-                rustls::TLSError::General(format!("Couldn't parse TEE report value: {:?}", error))
-            })?;
+                })?;
 
-        // Check that X.509 TEE extension contains a correct TEE report.
-        self.check_tee_report(tee_extension_value)
-            .map_err(|error| {
-                rustls::TLSError::General(format!("Incorrect TEE report: {:?}", error))
+            // Extract public key.
+            let parsed_openssl_certificate = openssl::x509::X509::from_der(&certificate.0)
+                .map_err(|error| {
+                    let error =
+                        rustls::TLSError::General(format!("Couldn't parse certificate: {}", error));
+                    warn!("{:?}", error);
+                    error
+                })?;
+            let public_key = parsed_openssl_certificate
+                .public_key()
+                .map_err(|error| {
+                    let error = rustls::TLSError::General(format!(
+                        "Couldn't extract public key: {}",
+                        error
+                    ));
+                    warn!("{:?}", error);
+                    error
+                })?
+                .public_key_to_pem()
```

Comment:
Would it not be better to use the raw public key bytes, or at least the DER encoded version? I believe PEM encoding can contain whitespace and still be valid, so there is no guarantee that the PEM encoding of the same bytes will be exactly the same for all libraries.